plugins/ecs-bridge/engine/engine.go

```diff
@@ -0,0 +1,245 @@
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package engine
+
+import (
+	"net"
+	"syscall"
+
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cniipamwrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cniipwrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/cninswrapper"
+	"github.com/aws/amazon-ecs-cni-plugins/pkg/netlinkwrapper"
+	"github.com/containernetworking/cni/pkg/types/current"
+	"github.com/pkg/errors"
+	"github.com/vishvananda/netlink"
+)
+
+// zeroLengthIPString is what we expect net.IP.String() to return if the
+// ip has length 0. We use this to determing if an IP is empty.
+// Refer https://golang.org/pkg/net/#IP.String
+const zeroLengthIPString = "<nil>"
```

Comment:
We can create a util method for it. I'll create an issue for it and address in a different PR since it's an enhancement anyway.